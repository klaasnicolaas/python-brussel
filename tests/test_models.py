"""Test the models."""
import aiohttp
import pytest
from aresponses import ResponsesMockServer

from brussel import DisabledParking, Garage, ODPBrussel

from . import load_fixtures


@pytest.mark.asyncio
async def test_all_garages(aresponses: ResponsesMockServer) -> None:
    """Test all garages function."""
    aresponses.add(
        "bruxellesdata.opendatasoft.com",
        "/api/records/1.0/search/",
        "GET",
        aresponses.Response(
            status=200,
            headers={"Content-Type": "application/json"},
            text=load_fixtures("garages.json"),
        ),
    )
    async with aiohttp.ClientSession() as session:
        client = ODPBrussel(session=session)
        spaces: list[Garage] = await client.garages()
        assert spaces is not None
        for item in spaces:
            assert isinstance(item, Garage)
            assert item.capacity is not None
            assert item.longitude is not None
            assert item.latitude is not None
            assert isinstance(item.longitude, float)
            assert isinstance(item.latitude, float)


@pytest.mark.asyncio
async def test_disabled_parkings(aresponses: ResponsesMockServer) -> None:
    """Test disabled parking spaces function."""
    aresponses.add(
        "bruxellesdata.opendatasoft.com",
        "/api/records/1.0/search/",
        "GET",
        aresponses.Response(
            status=200,
            headers={"Content-Type": "application/json"},
            text=load_fixtures("disabled_parkings.json"),
        ),
    )
    async with aiohttp.ClientSession() as session:
        client = ODPBrussel(session=session)
        spaces: list[DisabledParking] = await client.disabled_parkings()
        assert spaces is not None
        for item in spaces:
            assert item.spot_id is not None
            assert item.longitude is not None
            assert item.latitude is not None
            assert isinstance(item.longitude, float)
            assert isinstance(item.latitude, float)
