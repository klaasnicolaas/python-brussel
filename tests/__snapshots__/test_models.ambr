# serializer version: 1
# name: test_all_garages
  list([
    Garage(garage_id='29791f5b4a29f223213d6364c04194bf031bc829', name={'fr': 'City 2', 'nl': 'City 2'}, capacity=380, provider='Interparking', longitude=4.35906599521552, latitude=50.853812003718275, updated_at=datetime.datetime(2024, 2, 13, 17, 7, 13, 211000, tzinfo=datetime.timezone.utc)),
    Garage(garage_id='86c7c424dc328c853e18645de7c308feaa891059', name={'fr': 'Industrie', 'nl': 'Industrie'}, capacity=311, provider='Interparking', longitude=4.369964995943753, latitude=50.84289200411929, updated_at=datetime.datetime(2024, 2, 13, 17, 7, 13, 211000, tzinfo=datetime.timezone.utc)),
    Garage(garage_id='d1646967f3605ac66baff718c2bff21a3b7e7a89', name={'fr': 'Monnaie', 'nl': 'Munt'}, capacity=589, provider='Interparking', longitude=4.353903995527477, latitude=50.850452003526684, updated_at=datetime.datetime(2024, 2, 13, 17, 7, 13, 211000, tzinfo=datetime.timezone.utc)),
    Garage(garage_id='0ad4382c34684cc0957b2a93c5b037a5b41ae7da', name={'fr': 'Passage 44', 'nl': 'Passage 44'}, capacity=525, provider='Interparking', longitude=4.36149599502663, latitude=50.85251400351547, updated_at=datetime.datetime(2024, 2, 13, 17, 7, 13, 211000, tzinfo=datetime.timezone.utc)),
    Garage(garage_id='f9b674ef273644a1b9c91513cfa7693fe1325baf', name={'fr': 'Dansaert', 'nl': 'Dansaert'}, capacity=211, provider='Q-Park Belgium', longitude=4.345503995430967, latitude=50.851114004333084, updated_at=datetime.datetime(2024, 2, 13, 17, 7, 13, 211000, tzinfo=datetime.timezone.utc)),
    Garage(garage_id='9fcd69b3658512cd2564c58868eb7de61bafae29', name={'fr': 'De Brouckère', 'nl': 'De Brouckère'}, capacity=490, provider='Interparking', longitude=4.352112995676479, latitude=50.85091600413579, updated_at=datetime.datetime(2024, 2, 13, 17, 7, 13, 211000, tzinfo=datetime.timezone.utc)),
    Garage(garage_id='024e1b4e7c5655865f36cafb508c2fb48655caae', name={'fr': 'Deux Portes', 'nl': 'Deux Portes'}, capacity=929, provider='Interparking', longitude=4.361332996039703, latitude=50.83860100390401, updated_at=datetime.datetime(2024, 2, 13, 17, 7, 13, 211000, tzinfo=datetime.timezone.utc)),
    Garage(garage_id='4224625bd2b78ec0d6214c19abf5e316d538a5b8', name={'fr': 'Docks Bruxsel', 'nl': 'Docks Bruxsel'}, capacity=1670, provider='Indigo Belgium', longitude=4.3746399950728945, latitude=50.879780004069026, updated_at=datetime.datetime(2024, 2, 13, 17, 7, 13, 211000, tzinfo=datetime.timezone.utc)),
    Garage(garage_id='50d31a80a773c8e7a118fc4d2a43b6847fa0e613', name={'fr': 'Ecuyer', 'nl': 'Schildknaap'}, capacity=484, provider='Interparking', longitude=4.353418996124563, latitude=50.84929700393285, updated_at=datetime.datetime(2024, 2, 13, 17, 7, 13, 211000, tzinfo=datetime.timezone.utc)),
    Garage(garage_id='abec2e7b3d71e11c607ce71779e4f5e1488dbaef', name={'fr': 'Sablon-Poelaert', 'nl': 'Zavel-Poelaert'}, capacity=499, provider='Interparking', longitude=4.352839995462927, latitude=50.837956003675416, updated_at=datetime.datetime(2024, 2, 13, 17, 7, 13, 211000, tzinfo=datetime.timezone.utc)),
  ])
# ---
# name: test_disabled_parkings
  list([
    DisabledParking(spot_id='844c6302ac6fd8d391938e6004584d54dadd8573', number=2, orientation='Parallel', area={'en': 'Grey area', 'fr': 'Zone grise', 'nl': 'Grijze zone'}, longitude=4.355222056418672, latitude=50.838032395915626, updated_at=datetime.datetime(2024, 1, 15, 10, 55, 36, 165000, tzinfo=datetime.timezone.utc)),
    DisabledParking(spot_id='cc9bb841aa646fb4f7f8906ca6eb88a83cba9089', number=1, orientation='Parallel', area={'en': 'Green area', 'fr': 'Zone verte', 'nl': 'Groene zone'}, longitude=4.347506949332884, latitude=50.8742355167629, updated_at=datetime.datetime(2024, 1, 15, 10, 55, 36, 165000, tzinfo=datetime.timezone.utc)),
    DisabledParking(spot_id='b54c2339549bfb0baf1238b9b26ffc25945a46d3', number=1, orientation='Parallel', area={'en': 'Grey area', 'fr': 'Zone grise', 'nl': 'Grijze zone'}, longitude=4.3273906536192435, latitude=50.898033791939596, updated_at=datetime.datetime(2024, 1, 15, 10, 55, 36, 165000, tzinfo=datetime.timezone.utc)),
    DisabledParking(spot_id='85d2ab8df86c3bf372265afb338498ee2b2f5155', number=1, orientation='Parallel', area={'en': 'Grey area', 'fr': 'Zone grise', 'nl': 'Grijze zone'}, longitude=4.34488956916515, latitude=50.838435022680024, updated_at=datetime.datetime(2024, 1, 15, 10, 55, 36, 165000, tzinfo=datetime.timezone.utc)),
    DisabledParking(spot_id='78133f622dfe0c57ed1b3d7e2850baa693252c25', number=1, orientation='Parallel', area={'en': 'Grey area', 'fr': 'Zone grise', 'nl': 'Grijze zone'}, longitude=4.342811678108436, latitude=50.848906207806564, updated_at=datetime.datetime(2024, 1, 15, 10, 55, 36, 165000, tzinfo=datetime.timezone.utc)),
    DisabledParking(spot_id='8f2dd27c49910c906c4d312a23d95863e2361fef', number=1, orientation='Parallel', area={'en': 'Grey area', 'fr': 'Zone grise', 'nl': 'Grijze zone'}, longitude=4.346569491732328, latitude=50.85768790699441, updated_at=datetime.datetime(2024, 1, 15, 10, 55, 36, 165000, tzinfo=datetime.timezone.utc)),
    DisabledParking(spot_id='aec1ed45e54eacb6a96380555cce49e4c6994af7', number=1, orientation='Parallel', area={'en': 'Green area', 'fr': 'Zone verte', 'nl': 'Groene zone'}, longitude=4.356547607189667, latitude=50.87731935908454, updated_at=datetime.datetime(2024, 1, 15, 10, 55, 36, 165000, tzinfo=datetime.timezone.utc)),
    DisabledParking(spot_id='7cecd92eafc5f2ec7153104d7cd7f91b0cde5efe', number=2, orientation='Parallel', area={'en': 'Green area', 'fr': 'Zone verte', 'nl': 'Groene zone'}, longitude=4.360456339991621, latitude=50.87047820962347, updated_at=datetime.datetime(2024, 1, 15, 10, 55, 36, 165000, tzinfo=datetime.timezone.utc)),
    DisabledParking(spot_id='9a8e305d2d6cea03c6da6ff5d1744430c99586c1', number=1, orientation='Parallel', area={'en': 'Green area', 'fr': 'Zone verte', 'nl': 'Groene zone'}, longitude=4.326331727238244, latitude=50.8980741449515, updated_at=datetime.datetime(2024, 1, 15, 10, 55, 36, 165000, tzinfo=datetime.timezone.utc)),
    DisabledParking(spot_id='ae08e56f0c23560f74c3d857ff25d1d0fa6e286a', number=1, orientation='Parallel', area={'en': 'Blue area', 'fr': 'Zone bleue', 'nl': 'Blauwe zone'}, longitude=4.42036399758698, latitude=50.89519351159455, updated_at=datetime.datetime(2024, 1, 15, 10, 55, 36, 165000, tzinfo=datetime.timezone.utc)),
  ])
# ---
